### ISSUE TITLE
Wonderful work! But I don't know why can't the results be replicated with either the downloaded weights or the weights I trained myself?

### ISSUE THREAD

**Issue by supercyx3** [xiaozhen228/Bayes-PFL#13]
python -u train.py --dataset visa --train_data_path ./dataset/mvisa/data --val_data_path ./dataset/mvisa/data --save_path ./my_exps/train_visa --pretrained_path ./pretrai
ned_weight/ViT-L-14-336px.pt --prompt_context_len 5 --prompt_num 3 --prompt_state_len 5 --device_id 0 --learning_rate 0.0001 --features_list 6 12 18 24 --pretrained openai --image_size 518 --batch_size 32 --epochs 30 --sample_num 5 --num_flows 10 --config_path ./open_clip_local/model_configs/ViT-L-14-336.json --model ViT-L-14-336 --seed 333 --alpha 0.6
Segmentation Loss: 3.9353458881378174, Classification Loss: 0.3104017972946167, Distribution Loss: 68.75559997558594, Orthogonal_Loss: 1.2813684940338135
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 68/68 [12:24<00:00, 10.95s/it]
-------------------start evaluating ---------------------
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 547/547 [01:50<00:00,  4.93it/s] 
------------------- end evaluating ---------------------
25-08-26 21:05:14.980 - INFO: epoch post [16], loss_seg:4.4685 loss_cls:0.2726 loss_dist_reg:69.8993  loss_text:1.2717 ap_raw_pixel:0.5815  ap_raw_image:0.9793
25-08-26 21:05:14.980 - INFO: Group 0 LR: 0.000529
25-08-26 21:05:14.980 - INFO: Group 1 LR: 0.005291
25-08-26 21:05:14.981 - INFO: Group 2 LR: 0.000529
25-08-26 21:05:14.981 - INFO: Group 3 LR: 0.000053
25-08-26 21:05:15.110 - INFO: early stop

python -u test.py --dataset mvtec --data_path ./dataset/mvisa/data  --checkpoint_path D:\Bayes-PFL-main\Bayes-PFL-main\my_exps\train_visa\epoch_post_16_2.pth  --save_path ./results/test_visa --pretrained_path ./pretrained_weight/ViT-L-14-336px.pt  --prompt_context_len 5 --prompt_num 3 --prompt_state_len 5 --device_id 0 --features_list 6 12 18 24 --pretrained openai --image_size 518  --seed 333 --config_path ./open_clip_local/model_configs/ViT-L-14-336.json --model ViT-L-14-336  --sample_num 10 --num_flows 10
| objects    |   auroc_px |   aupro_px |   ap_px |   f1_px |   iou |   auroc_sp |   aupro_sp |   ap_sp |   f1_sp |   threshold |
|:-----------|-----------:|-----------:|--------:|--------:|------:|-----------:|-----------:|--------:|--------:|------------:|
| bottle     |      93.66 |          0 |   66.32 |   62.74 | 45.71 |      96.98 |          0 |   99.12 |   95.38 |       0.581 |
| cable      |      78.41 |          0 |   14.8  |   22.67 | 12.78 |      86.24 |          0 |   92.26 |   83.72 |       0.696 |
| capsule    |      95.98 |          0 |   29.26 |   34.34 | 20.73 |      93.26 |          0 |   98.62 |   93.81 |       0.638 |
| carpet     |      99.43 |          0 |   81.6  |   74.73 | 59.66 |     100    |          0 |  100    |  100    |       0.494 |
| grid       |      97.79 |          0 |   39.64 |   42.28 | 26.8  |      99.83 |          0 |   99.94 |   99.13 |       0.561 |
| hazelnut   |      97.17 |          0 |   50.42 |   50.51 | 33.79 |      93.18 |          0 |   96.45 |   91.16 |       0.778 |
| leather    |      99.58 |          0 |   59.28 |   57.31 | 40.16 |     100    |          0 |  100    |  100    |       0.66  |
| metal_nut  |      75.35 |          0 |   25.13 |   34.72 | 21.01 |      71.21 |          0 |   93.26 |   89.42 |       0.355 |
| pill       |      86.67 |          0 |   31.88 |   36.4  | 22.25 |      91    |          0 |   98.18 |   93.91 |       0.609 |
| screw      |      98.77 |          0 |   43.31 |   46.27 | 30.1  |      91.33 |          0 |   97.03 |   90.62 |       0.769 |
| tile       |      94.62 |          0 |   74.39 |   70.94 | 54.97 |      99.28 |          0 |   99.72 |   97.62 |       0.464 |
| toothbrush |      94.01 |          0 |   32.95 |   35.73 | 21.75 |      94.44 |          0 |   97.93 |   93.55 |       0.717 |
| transistor |      72.03 |          0 |   11.93 |   17.22 |  9.42 |      73.04 |          0 |   71.51 |   65.26 |       0.522 |
| wood       |      96.39 |          0 |   67.44 |   64.31 | 47.4  |      99.74 |          0 |   99.92 |   98.36 |       0.524 |
| zipper     |      96.9  |          0 |   62.6  |   60.5  | 43.37 |      97.87 |          0 |   99.42 |   96.64 |       0.486 |
| mean       |      91.79 |          0 |   46.06 |   47.38 | 32.66 |      92.49 |          0 |   96.22 |   92.57 |       0.59  |

**Comment by xiaozhen228** [comment]
@supercyx3  Hello, I noticed that you seem to have used a Windows system for training. I am not sure if the difference comes from the platform, as I used a Linux system. Secondly, since our method involves randomness and uncertainty during sampling, it is normal for inference results to differ when using the uploaded weights. Lastly, from your experiment logs, I observed that ap_raw_pixel: 0.5815 is slightly lower; normally, the optimal performance on the validation set should remain around ap_max_pixel update: 0.60. Therefore, I suggest keeping your experimental environment consistent with ours or trying to adjust the network hyperparameters (e.g., learning rate) and random seed.
